lib/system_commands.rb

```diff
@@ -32,8 +33,14 @@ def sh!(*args)
       end
     end
 
+    unless options[:timeout].nil?
+      cmd = "timeout #{options[:timeout]} #{cmd}"
+    end
+
     output = `#{cmd} 2>&1`
     status = $?
+
+    fail "Time out for command ''#{cmd}'', status #{status.inspect}. The output follows:\n#{output}" if status.exitstatus == 124
```

Comment:
I'd like to nitpick the wording a bit. How about:

    "Command `#{cmd}` timed out, status ..."